src/WebJobs.Script.Grpc/Proto/FunctionRpc.proto

```diff
@@ -0,0 +1,255 @@
+  // i.e. Worker = Verbose, Function.MyFunc = None
+  map<string, RpcLog.Level> log_categories = 3;
+}
+
+message WorkerInitResponse {
+  string worker_version = 1;
+  // A map of worker supported features/capabilities
+  map<string, string> capabilities = 2;
+  StatusResult result = 3;
+}
+
+message StatusResult {
+  enum Status {
+    Failure = 0;
+    Success = 1;
+    Cancelled = 2;
+  }
+  Status status = 4;
+  string result = 1;
+  RpcException exception = 2;
+  // Captured logs or relevant details can use the logs property
+  repeated RpcLog logs = 3;
+}
+
+// TODO: investigate grpc heartbeat - don't limit to grpc implemention
+// Message is empty by design - Will add more fields in future if needed
+message WorkerHeartbeat {}
+
+// Warning before killing the process after grace_period
+// Worker self terminates ..no response on this
+message WorkerTerminate {
+  google.protobuf.Duration grace_period = 1;
+}
+
+message FileChangeEventRequest {
+  // https://msdn.microsoft.com/en-us/library/t6xf43e0(v=vs.110).aspx
+  enum Type {
+	Unknown = 0;
+    Created = 1;
+    Deleted = 2;
+    Changed = 4;
+    Renamed = 8;
+    All = 15;
+  }
+
+  Type type = 1;
+  string full_path = 2;
+  string name = 3;
+}
+
+message WorkerActionResponse {
+  enum Action {
+    Restart = 0;
+    Reload = 1;
+  }
+  
+  Action action = 1;
+  string reason = 2;
+}
+
+message WorkerStatusRequest{
+}
+
+message WorkerStatusResponse {
+}
+
+message FunctionLoadRequest {
+  // unique function identifier (avoid name collisions, facilitate reload case)
+  string function_id = 1;
+  RpcFunctionMetadata metadata = 2;
+}
+
+message FunctionLoadResponse {
+  string function_id = 1;
+  StatusResult result = 2;
+  // TODO: return type expected?
+}
+
+message RpcFunctionMetadata {
+  // TODO: do we want the host's name - the language worker might do a better job of assignment than the host
+  string name = 4;
+
+  string directory = 1;
+  string script_file = 2;
+  string entry_point = 3;
+
+  map<string, BindingInfo> bindings = 6;
+
+  // not adding disabled or excluded as those (currently) are only relevant to host
+}
+
+message InvocationRequest {
+  string invocation_id = 1;
+  string function_id = 2;
+  repeated ParameterBinding input_data = 3;
+  map<string, TypedData> trigger_metadata = 4;
+}
+
+message InvocationCancel {
+  string invocation_id = 2;
+  google.protobuf.Duration grace_period = 1; // could also use absolute time
+}
+
+message InvocationResponse {
+  string invocation_id = 1;
+  repeated ParameterBinding output_data = 2;
+  TypedData return_value = 4;
+  StatusResult result = 3;
+}
+
+message TypedData {
+  oneof data {
+    string string = 1;
+    string json = 2;
+    bytes bytes = 3;
+    bytes stream = 4;
+    RpcHttp http = 5;
+	sint64 int = 6;
+    double double = 7;
+  }
+}
+
+message ParameterBinding {
+  string name = 1;
+  TypedData data = 2;
+}
+
+message BindingInfo {
+  string type = 2;
```

Comment:
Do we actually need this? Can we remove it?

Reply:
Currently yes, due to http return. Will investigate if this can be moved to script layer https://github.com/Azure/azure-webjobs-sdk-script/issues/1836